### PR TITLE
Try setting up CI

### PR DESCRIPTION
**Pull request by sampsyo**
A totally not ready, experimental approach to #53. Nothing to see here yet.

### CODE REVIEW

**Comment by sampsyo** [comment]
One note as this starts to take shape: so far, this only runs the tests for slow-odgi. It should eventually also run the tests for the data generator and the new tests for the parser added in #136.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+on:
+  push:
+    branches:
+     - main
+  pull_request:
+    branches:
+     - main
+
+jobs:
```
Comment (anshumanmohan) [inline .github/workflows/build.yml:11]: Nice, this feels a lot like our Dockerfile. Which is a good thing. In fact maybe in the future the Dockerfile should learn from your system of sourcing and running odgi.

Comment (sampsyo) [reply .github/workflows/build.yml:11]: Yeah, totally. I stole some of the stuff directly from the Dockerfile. Someday merging these things could be cool, but I have never really figured out the way to do this in general for other projects (i.e., to use a "base" Dockerfile twice, once for the "real" Dockerfile and once for the CI).

**Comment by sampsyo** [comment]
Excellent; merging this now!

**Pull request merged**